libvdsk/vdsk_int.h

```diff
@@ -30,6 +30,7 @@
 #define	__VDSK_INT_H__
 
 #include <sys/linker_set.h>
+#include <pthread.h>
```

Comment:
I think we should be able to build without libpthread as well as with libpthread. For a non-threaded application that uses libvdsk, we don't need locking. For a threaded application (like bhyve), we do. Maybe build 2 variants of libvdsk?

Reply:
Would it be a good idea to guard it with an #ifdef MULTITHREAD, or something like that?

Reply:
Yeah, something like that. Let's figure that out later (i.e. after the pull).